tests/worklist_server_tests.py

```diff
@@ -90,34 +91,181 @@ def test_RequireThat_WorklistRequest_ReturnsWorklistWithAllExpectedFieldsPopulat
+        self._server._config_values.reset()
 
-        self.assertEqual(worklist1, worklist2)
+    def test_RequireThat_WorklistRequest_ReturnsLongStrings_WhenSet(self):
+        self._server._config_values.reset()
+        self._server._config_values.rateOfCleanExams = 0
+        self._server._config_values.rateOfRandomExams = 2
 
+        self._server._config_values.long_enabled = 1
+        self._server._config_values.empty_enabled = 0
+
+        self._server._config_values.likelyhood_of_long_string = 100.0
 
-    def test_RequreThat_WorklistRequest_ReturnDifferentList_WhenSeedNotSpecified(self):
-        self._setUpTest(0)
         client = worklist_client.WorklistClient(self._server_config.network_address)
 
-        query_dataset1 = Dataset()
-        query_dataset1.PatientName = '*'
-        worklist1 = client.get_worklist(query_dataset1)
+        query_dataset = Dataset()
+        query_dataset.PatientName = '*'
+
+
+        worklist = client.get_worklist(query_dataset)
+        worklist_item = worklist[0]
+
+        self.assertTrue(len(worklist_item.CurrentPatientLocation) > 64)
+
+        self._server._config_values.long_enabled = 0
+
+        worklist = client.get_worklist(query_dataset)
+        worklist_item = worklist[0]
+
+        self.assertTrue(len(worklist_item.CurrentPatientLocation) <= 64)
+
+        self._server._config_values.reset()
+
+    def test_RequireThat_WorklistRequest_ReturnsEmptyStrings_WhenSet(self):
+        self._server._config_values.reset()
+        self._server._config_values.rateOfCleanExams = 0
+        self._server._config_values.rateOfRandomExams = 2
+
+        self._server._config_values.long_enabled = 0
+        self._server._config_values.empty_enabled = 1
+
+        self._server._config_values.likelyhood_of_long_string = 0.0
+        self._server._config_values.likelyhood_of_empty_string = 100.0
+
+        client = worklist_client.WorklistClient(self._server_config.network_address)
+
+        query_dataset = Dataset()
+        query_dataset.PatientName = '*'
+
+        worklist = client.get_worklist(query_dataset)
+        worklist_item = worklist[0]
+
+        self.assertEqual(len(worklist_item.CurrentPatientLocation), 0)
+
+        self._server._config_values.reset()
+
+
+
+    def test_RequireThat_MultipleWorklistRequests_ReturnsIdenticalWorklists_ByUseOfSeed(self):
+        client = worklist_client.WorklistClient(self._server_config.network_address)
+
+        query_dataset = Dataset()
+        query_dataset.PatientName = '*'
+
+        seed = random.randrange(1000)
+
+        random.seed(seed)
+        worklist_one = client.get_worklist(query_dataset)
+
+        random.seed(seed)
+        worklist_two = client.get_worklist(query_dataset)
+
+        exam = random.randrange(len(worklist_one))
+
+        worklist_item_one = worklist_one[exam]
+        worklist_item_two = worklist_two[exam]
+
+        self.assertEqual(worklist_item_one.StudyInstanceUID, worklist_item_two.StudyInstanceUID)

```

Comment:
Here, you can reduce duplication in code by first creating a list of strings containing the tag names. Then, you can iterate over the datasets like this:

    for tag in tags:
        self.assertEqual(worklist_item_one[tag], worklist_item_two[tag]